rust-runtime/aws-smithy-runtime-api/src/client/runtime_plugin.rs

```diff
@@ -235,21 +235,13 @@ impl RuntimePlugins {
         Default::default()
     }
 
-    pub fn with_client_plugin(mut self, plugin: impl RuntimePlugin + 'static) -> Self {
-        insert_plugin!(
-            self.client_plugins,
-            plugin,
-            SharedRuntimePlugin::new(plugin)
-        );
+    pub fn with_client_plugin(mut self, plugin: SharedRuntimePlugin) -> Self {
+        insert_plugin!(self.client_plugins, plugin);
         self
     }
 
-    pub fn with_operation_plugin(mut self, plugin: impl RuntimePlugin + 'static) -> Self {
-        insert_plugin!(
-            self.operation_plugins,
-            plugin,
-            SharedRuntimePlugin::new(plugin)
-        );
+    pub fn with_operation_plugin(mut self, plugin: SharedRuntimePlugin) -> Self {
+        insert_plugin!(self.operation_plugins, plugin);
         self
     }
```

Comment:
seems like this makes the API a bit more annoying in a bunch of places—I don't necessarily mind making the API tighter here but maybe we could add an extension trait or something?

Reply:
Fixed in 01720a953b12fc775e3f0c37d83d6f6887bba4b8